src/api/route-services/pipeline-response.js

```diff
@@ -78,9 +80,54 @@ const pipelineResponse = async (io, message) => {
     ]);
   }
 
-
   const statusRes = await pipelineStatus(experimentId);
 
+  if (taskName === 'configureEmbedding') {
+    const now = new Date();
+    const addTimeoutSeconds = 60;
+    const timeout = new Date(now.getTime() + addTimeoutSeconds * 1000);
+
+    // Run work request for embedding
+    const embeddingWorkRequest = {
+      experimentId,
+      timeout,
+      uuid: uuidv4(),
+      body: {
+        name: 'GetEmbedding',
+        type: output.config.embeddingSettings.method,
+        config: output.config.embeddingSettings.methodSettings[
+          output.config.embeddingSettings.method
+        ],
+      },
+      PipelineRunETag: statusRes.pipeline.startDate,
+    };
+
+    const embeddingWork = new WorkSubmitService(embeddingWorkRequest);
+    await embeddingWork.submitWork();
+
+    // Run work request for cell clustering
+    const clusteringWorkRequest = {
+      experimentId,
+      timeout,
+      uuid: uuidv4(),
+      body: {
+        name: 'ClusterCells',
+        cellSetName: 'Louvain clusters',
+        type: output.config.clusteringSettings.method,
+        cellSetKey: 'louvain',
+        config: {
+          resolution: output.config.clusteringSettings.methodSettings[
+            output.config.clusteringSettings.method
+          ],
+        },
+      },
+      PipelineRunETag: statusRes.pipeline.startDate,
+    };
```

Comment:
same here - this piece of code is easy to break -- all it takes is to do a tiny change in the task schema.

Reply:
This is refactored into `hookRunner` and associated functions under `src/utils/hooks`